Switch to STIX2 and vendor it

Toxaris opened the pull request:
Hopefully helpful on Mac

Toxaris committed:
Add fontspec files for XITS

Hopefully helpful on Mac

skaldarnar commented:
This alone does not help for me on Mac. 

Maybe we should vendor the font like we do with the other fonts?

Toxaris committed:
Switch to STIX2 and vendor it

Toxaris commented:
I now vendored all fonts as proposed by @skaldarnar. When looking for the file, I noted that XITS is deprecated in favor of STIX2 so I updated to that.

BTW I first tried to put the files into a subfolder, but I couldn't get that to work. So I moved everything back to the root folder and now it works again on my Windows machine and in the Github action :)

@skaldarnar or @fkz: can you check whether it works for you on Mac now?

skaldarnar commented:
Works for me! 👍
